src/Server/Workspace.php

```diff
@@ -216,28 +216,55 @@ public function xdependencies(): array
     }
 
     /**
-     * @param Options|null $settings
+     * Fires when client changes settings in the client
+     *
+     * The default paramter type is Options but it also accepts different types
+     * which will be transformed on demand.
+     *
+     * Currently only the vscode format is supported
+     *
+     * @param mixed|null $settings
+     * @return bool
+     * @throws \Exception Settings format not valid
      */
-    public function didChangeConfiguration(Options $settings = null)
+    public function didChangeConfiguration($settings = null): bool
     {
+        // List of options that affect the indexer
+        $indexerOptions = ['fileTypes'];
+
         if ($settings === null) {
-            return;
+            return false;
+        }
+
+        // VSC sends the settings with the config section as main key
+        if ($settings instanceof \stdClass && $settings->phpIntelliSense) {
+            $mapper = new \JsonMapper();
+            $settings = $mapper->map($settings->phpIntelliSense, new Options);
```

Comment:
Sure we can.

We only have to lookout for possible conflicts in the future when we use `php` as config section.

Possible config naming:

- `php.indexer.*`
- `php.completion.*` (possible name for a feature request, but can be ignored now)
- `php.intellisense.*` (previous name)